### ISSUE TITLE
display-name false negative with React.createClass and ES6-style method

### ISSUE THREAD

**Issue by peet** [yannickcr/eslint-plugin-react#852]
I initially raised #650 because I had many components not reporting display-name issues when being assigned directly to module.exports (and ending up with 'exports' as the name in the devtools).

I've since done some for investivation and narrowed it down to using React.createClass() passing in an object containing an ES6-style method.

This does not report an error for `"react/display-name": "error"`
```javascript
var React = require('react');
module.exports = React.createClass({
  render() {
    return null;
  }
});
```
or even:
```javascript
var React = require('react');
module.exports = React.createClass({
  method() {},
  render: function() {
    return null;
  }
});
```

Further, it has something to do with the method() property having its own 'name', not the whole class, as this also does not trigger the display-name error:
```javascript
var React = require('react');
module.exports = React.createClass({
  method: function method() {},
  render: function() {
    return null;
  }
});
```
and
```javascript
var React = require('react');
module.exports = React.createClass({
  render: function named() {
    return null;
  }
});
```

`eslint-pluign-react: "6.2.2"`

example repo [peet/display-name-test](https://github.com/peet/display-name-test)
